### PR TITLE
Commit the updated renv.lock files during deployment

### PR DESCRIPTION
**Pull request by vedhav**
### Current Issues
1. There was a bug in the logic of committing the renv.lock files - Even though the deployment and local running of the apps will use the latest NEST packages (stable/deval) due to the non-updated renv.lock files it might seem that we are using older NEST packages.
2. We would also need to update the `activate.R` file because updates to the `renv` package usually come with a new activate file.

### Doubts
1. Is it possible to directly commit these lock files on `main` and `dev`? I'm worried of some permission issues.
2. Is it possible to trigger an automation that will merge all the code files from the `main` branch to `dev` branch when someone merges to `main`? Except for merging the `renv.lock` file and `renv` directory from the `main` branch.

### Pre-merge
Make sure to replace the names of the branches from `update-renv-lock@main` to `main` and `update-renv-lock@dev` to `dev`. And remove the push trigger from the `update-renv-lock@main` branch.

### CODE REVIEW

**Comment by vedhav** [comment]
> That depends. What's your branching strategy here? Are dev and main divergent?

Yes, They can only diverge in the file named `renv.lock` inside the app directories like `basic-teal/renv.lock` and the `renv` directories inside the app directories like `basic-teal/renv/`. But all other things from the `main` should automatically be merged with `dev` when a main merge occurs.

**Comment by vedhav** [comment]
> Why create a separate branch for dev content, though? Why not have a nested directory/file structure where dev 
>  environments are separate from their stable counterparts?

This sounds like a good solution. Now that you mention this I see there is another problem that having a separate `dev` branch solves which is why we created this in the first place - If there are some breaking changes (like a refactor that affects the teal app developers) the app code can start to deviate and we need to have the flexibility to change the app code and test it in the `dev` branch with the "main-branch version" of the packages. But now that I think about this problem I don't think merging `main` to `dev` is something that we should do.
I think we should just start following a workflow that the code will always be squash-merged with the `dev` first and when the NEST packages are released we can merge the "app code" (not `renv`-related stuff) to the `main` (triggered manually as package release might take time). Is it possible to do this "app code only" merge? The only thing that comes to my mind is to create a merge action that is triggered using dispatch manually when such merge from dev is needed, And I think a similar `main` to `dev` code merge should also be possible.

I'll think some more about this. Thanks for a productive retro ;)

**Comment by m7pr** [comment]
Hey @cicdguy @vedhav did you had a meeting in the end? What were conclusions?
We can either have a main and dev branch or main branch with 2 directories but I think it's better to have 2 branches so that the comparison and the review is possible. I would advice on merging to dev first, and then to master after testing and the reviews. This is the regular strategy where dev stands for things being currently developer and main for things being tested and stable. 

I'm not that sure we should have plenty of automated jobs that merge changes automatically to main. Teal Gallery is our main showcase with the biggest exposure, I suggest reviewing all changes and deployments manually before they go live publically. The worst thing we want is to get 9 commits per day being pushed to master without any actions on testing the stability of the main branch, hence the deployed versions of showcase apps. I would actually opt for a stabilziation period, where for the next 6 months we do verify apps manually before each publication.

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
   push:
     branches:
       - main
+      - update-renv-lock@main
```
Comment (m7pr) [inline .github/workflows/deploy.yaml:8]:
```suggestion
```

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -48,13 +49,15 @@ jobs:
...
       - name: Checkout repo 🛎
         uses: actions/checkout@v3
+        with:
+          ref: ${{ matrix.channel == 'stable' && 'update-renv-lock@main' || 'update-renv-lock@dev' }} # TODO: Change the ref with main and dev
 
       - name: Set channel related constants
         run: |
           if [ "${{ matrix.channel }}" = "stable" ]; then
-            echo "BRANCH_NAME=main" >> $GITHUB_ENV # TODO: Change the branch name to main
+            echo "BRANCH_NAME=update-renv-lock@main" >> $GITHUB_ENV # TODO: Change the branch name to main
```
Comment (m7pr) [inline .github/workflows/deploy.yaml:58]:
```suggestion
            echo "BRANCH_NAME=main" >> $GITHUB_ENV
```

File: .github/workflows/deploy.yaml [hunk 3]
```diff
@@ -48,13 +49,15 @@ jobs:
     steps:
       - name: Checkout repo 🛎
         uses: actions/checkout@v3
+        with:
+          ref: ${{ matrix.channel == 'stable' && 'update-renv-lock@main' || 'update-renv-lock@dev' }} # TODO: Change the ref with main and dev
```
Comment (m7pr) [inline .github/workflows/deploy.yaml:53]:
```suggestion
          ref: ${{ matrix.channel == 'stable' && 'main' || 'dev' }} 
```

File: .github/workflows/deploy.yaml [hunk 4]
```diff
@@ -48,13 +49,15 @@ jobs:
...
 
       - name: Set channel related constants
         run: |
           if [ "${{ matrix.channel }}" = "stable" ]; then
-            echo "BRANCH_NAME=main" >> $GITHUB_ENV # TODO: Change the branch name to main
+            echo "BRANCH_NAME=update-renv-lock@main" >> $GITHUB_ENV # TODO: Change the branch name to main
           else
-            echo "BRANCH_NAME=dev" >> $GITHUB_ENV # TODO: Change the branch name to deval
+            echo "BRANCH_NAME=update-renv-lock@dev" >> $GITHUB_ENV # TODO: Change the branch name to dev
```
Comment (m7pr) [inline .github/workflows/deploy.yaml:60]:
```suggestion
            echo "BRANCH_NAME=dev" >> $GITHUB_ENV
```

**Review by m7pr — CHANGES_REQUESTED** [review]
As @vedhav stated in his original comment

> Make sure to replace the names of the branches from update-renv-lock@main to main and update-renv-lock@dev to dev. And remove the push trigger from the update-renv-lock@main branch.

Putting that as a review so it's visible and required

**Comment by vedhav** [comment]
> I suggest reviewing all changes and deployments manually before they go live publically.

True, this sounds like a nice strategy to deploy our gallery. But, the additional use case served by teal.gallery is to make sure that the latest CRAN packages or our own changes don't break the apps. Re-deploying with updated packages is the only way to ensure this. Unless we can implement this test elsewhere we will not be able to manually deploy.

**Comment by m7pr** [comment]
Yeah, and that what dev environemnt should be for. Not main environemnt : )
You automatically deploy to dev, you make manual testing, if evertyhuing is fine, you deploy to main.

**Pull request closed without merging**